doc/dts/device_tree.rst

```diff
@@ -0,0 +1,208 @@
+describe Zephyr specific configuration information.  The Zephyr specific
+information is intended to augment the device tree descriptions.  The main
+reason for this is to leverage existing device tree files that a SoC vendor may
+already have defined for a given platform.
+
+Today, configuration in Zephyr comes from a number of different places.  It can
+come from Kconfig files, CMSIS header files, vendor header files, prj.conf
+files, and other miscellaneous sources.  The intent of using device tree is to
+replace or curtail the use of Kconfig files throughout the system, and instead
+use device tree files to describe the configuration of device nodes.  CMSIS and
+vendor header files can be used in conjunction with the device tree to fully
+describe hardware.  Device tree is not intended to replace CMSIS or vendor
+include files.
+
+During compilation the device tree files are compiled using the device tree
+compiler.  The compiler will run the .dts file through the C preprocessor to
+resolve any macro or #defines utilized in the file.  The output of the compile
+is another dts formatted file.
+
+After compilation, a python script extracts information from the compiled device
+tree file using a set of rules specified in YAML files.  The extracted
+information is placed in a include file and this include information is used by
+the rest of the code as the project is compiled.
+
+A temporary fixup file is required for device tree support on most devices.
+This .fixup file resides in the dts architecture directory and has the same
+name as the master .dts file.  The only difference is the suffix is .fixup.
+This fixup file maps the generated include information to the current
+driver/source usage.
+
+Device tree file formats
+************************
+Hardware and software is described inside of device tree files in clear text format.
+These files have the file suffix of .dtsi or .dts.  The .dtsi files are meant to
+be included by other files.  Typically for a given board you have some number of
+.dtsi include files that pull in common device descriptions that are used across
+a given SoC family.
+
+Example: FRDM K64F Board and Hexiwear K64
+Both of these boards are based on the same NXP Kinetis SoC family, the K6X.  The
+following shows the include hierarchy for both boards.
+
+dts/arm/frdm_k64.dts includes the following two files::
+
+  dts/arm/nxp/nxp_k6x.dtsi
+  dts/arm/armv7-m.dtsi
+
+dts/arm/hexiwear_k64.dts includes the same two files::
+
+  dts/arm/nxp/nxp_k6x.dtsi
+  dts/arm/armv7-m.dtsi
+
+The board specific .dts files enable nodes, defines the zephyr specific items
+and also adds board specific changes like gpio/pinmux assignments.  These types
+of things will vary based on the board layout and application use.
+
+Currently supported boards
+**************************
+
+Device tree is currently supported on all ARM targets.  Support for all other
+architectures is to be done in the near future.
```

Comment:
I'd drop the "in the near future" since this isn't defined (is there a planned release for full DTS support?)